### PR TITLE
Rollback changes from PR #657

### PR DESCRIPTION
**Pull request by laszlof**
This change rolls back a bug fix from PR #657. This PR contained no test's, and no real documentation what the issue was it was trying to solve. As a result, this change has broken updates (amongst other things) on at least anyone using curl 7.29.

See Issue #838 for more details.

## Pull Request Type

- [X] I have checked there is no other PR open for the same change.

This is a:
- [X] Bug fix
- [ ] New feature
- [ ] Documentation improvement
- [ ] Code quality improvement

## Context
Roll back a bug fix from PR #657. 

## Quality assurance

- [X] This change does NOT contain a breaking change (fix or feature that would cause existing functionality to change).
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added unit tests to accompany this PR.
- [ ] The (new/existing) tests cover this PR 100%.
- [X] I have (manually) tested this code to the best of my abilities.
- [X] My code follows the style guidelines of this project.

## Documentation

For new features:
- [ ] I have added a code example showing how to use this feature to the [`examples`](https://github.com/WordPress/Requests/tree/develop/examples) directory.
- [ ] I have added documentation about this feature to the [`docs`](https://github.com/WordPress/Requests/tree/develop/docs) directory.
    If the documentation is in a new markdown file, I have added a link to this new file to the Docs folder [`README.md`](https://github.com/WordPress/Requests/tree/develop/docs/README.md) file.

### CODE REVIEW

**Review by schlessera — APPROVED** [review]

**Comment by jrfnl** [comment]
For the record: the underlying issue #656 contains information on what #657 solves.

File: src/Transport/Curl.php [hunk 1]
```diff
@@ -383,7 +382,7 @@ private function setup_handle($url, $headers, $data, $options) {
 		$options['hooks']->dispatch('curl.before_request', [&$this->handle]);
 
 		// Force closing the connection for old versions of cURL (<7.22).
```
Comment (sybrew) [inline src/Transport/Curl.php:384]: This comment is no longer valid.

Comment (schlessera) [reply src/Transport/Curl.php:384]: The comment was already present before we made the problematic change and was the reason why we used the specific version constraint we ended up with. Before we fully understand the context and where this was originally coming for, I'll prefer to keep in in there.

Comment (tomsommer) [reply src/Transport/Curl.php:384]: The version number in the comment could be a cause for confusion going forward, just a heads up.

Comment (schlessera) [reply src/Transport/Curl.php:384]: Yes, indeed, and I think that's what has already happened in this case.
But I'm still assuming there is a reason why this was put in there initially. I'd like to understand this better before removing it.

Comment (schlessera) [reply src/Transport/Curl.php:384]: For future reference, this is the PR that introduced the comment: https://github.com/WordPress/Requests/pull/211

**Pull request merged**